network/common/src/main/java/org/apache/spark/network/server/OneForOneStreamManager.java

```diff
@@ -109,15 +111,34 @@ public void connectionTerminated(Channel channel) {
     }
   }
 
+  @Override
+  public void checkAuthorization(TransportClient client, long streamId) {
+    if (client.getClientId() != null) {
```

Comment:
I'm confused about this: since `getClientId` returns null if the client did not enable `spark.authenticate`, does that mean any application that did not enable SASL can read my shuffle files?